### PR TITLE
move queries in service to dataSource

### PR DESCRIPTION
**Pull request by Asiasama710**

### CODE REVIEW

File: app/core/src/main/java/com/thechance/core/data/datasource/CategoryDataSourceImp.kt [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+                    quantity = productRow[ProductTable.quantity],
+                )
+            }
+    }
+
+    override suspend fun isCategoryDeleted(categoryId: Long): Boolean = dbQuery {
+        val category = CategoriesTable.select { CategoriesTable.id eq categoryId }.singleOrNull()
+        category?.let {
+            it[CategoriesTable.isDeleted]
+        } ?: throw IdNotFoundException()
```
Comment (ahmednasserzaza) [inline app/core/src/main/java/com/thechance/core/data/datasource/CategoryDataSourceImp.kt]: We don't need to throw exception here

File: app/core/src/main/java/com/thechance/core/data/datasource/CategoryDataSourceImp.kt [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    }
+
+    override suspend fun isMarketDeleted(marketId: Long): Boolean = dbQuery {
+        val market = MarketTable.select { MarketTable.id eq marketId }.singleOrNull()
+        market?.let {
+            it[MarketTable.isDeleted]
+        } ?: throw NoSuchElementException()
```
Comment (ahmednasserzaza) [inline app/core/src/main/java/com/thechance/core/data/datasource/CategoryDataSourceImp.kt]: and here

File: app/core/src/main/java/com/thechance/core/data/datasource/MarketDataSourceImp.kt [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+            }
+            Market(marketId = marketId, marketName = marketName)
+        }
+
+    override suspend fun isDeleted(marketId: Long): Boolean =
+        dbQuery {
+            val market = MarketTable.select { MarketTable.id eq marketId }.singleOrNull()
+            market?.let {
+                it[MarketTable.isDeleted]
+            } ?: throw IdNotFoundException()
```
Comment (ahmednasserzaza) [inline app/core/src/main/java/com/thechance/core/data/datasource/MarketDataSourceImp.kt]: throw exception should be in use case not data source

**Review by ahmednasserzaza — COMMENTED** [review]
Well done , but what about dao that has all Queries , and data source should call that dao and use fubction that make queres and return result using dependency inversion , That's what I think

File: app/core/src/main/java/com/thechance/core/data/datasource/CategoryDataSourceImp.kt [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+        }
+        Category(
+            categoryId = newCategory[CategoriesTable.id].value,
+            categoryName = newCategory[CategoriesTable.name].toString(),
+            imageId = newCategory[CategoriesTable.imageId]
+        )
+    }
+
+    override suspend fun getCategoriesByMarketId(marketId: Long?): List<Category> = dbQuery {
+        CategoriesTable.select {
```
Comment (NadaFeteiha) [inline app/core/src/main/java/com/thechance/core/data/datasource/CategoryDataSourceImp.kt]: remove null "?", market Id will never be null here.

File: app/core/src/main/java/com/thechance/core/data/datasource/CategoryDataSourceImp.kt [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+            CategoriesTable.marketId eq marketId and
+                    CategoriesTable.isDeleted.eq(false)
+        }.map { resultRow ->
+            Category(
+                categoryId = resultRow[CategoriesTable.id].value,
```
Comment (NadaFeteiha) [inline app/core/src/main/java/com/thechance/core/data/datasource/CategoryDataSourceImp.kt]: you can use a mapper. toCategory.

File: app/core/src/main/java/com/thechance/core/data/datasource/CategoryDataSourceImp.kt [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+        }
+        true
+    }
+
+    override suspend fun getAllProductsInCategory(categoryId: Long?): List<Product> = dbQuery {
+        (ProductTable innerJoin CategoryProductTable)
+            .select { CategoryProductTable.categoryId eq categoryId }
+            .map { productRow ->
+                Product(
+                    id = productRow[ProductTable.id].value,
```
Comment (NadaFeteiha) [inline app/core/src/main/java/com/thechance/core/data/datasource/CategoryDataSourceImp.kt]: use mapper toProduct

**Review by NadaFeteiha — CHANGES_REQUESTED** [review]

**Review by NadaFeteiha — APPROVED** [review]

**Pull request merged**